### DISCUSSION TITLE
Fetch a remote OpenAPI file with Authorization headers?

### DISCUSSION THREAD

**Discussion by florianmartens** [openapi-ts/openapi-typescript#1886] [Q&A] [answered]
Thanks for this cool project.

The docs say that we can access a remote open-api.yaml like this:

`npx openapi-typescript https://myapi.dev/api/v1/openapi.yaml -o ./path/to/my/schema.d.ts`

But let's assume this file was behind an auth check and I would need to send an `Authorization: Bearer xxx` header. How could i achieve this? Is it possible?

Thanks.

**Comment by drwpow** [comment, marked as answer]
https://openapi-ts.dev/cli#auth

**Reply by florianmartens** [reply]
Awesome! Sorry for missing that. I'm not quite sure about the connection between this library and redocly. Is this project managed/owned by Redocly? I guess there's no way to pass fetch arguments to the CLI directly without a `redocly.yaml`?

**Reply by drwpow** [reply]
Nope, unrelated! The reason for merging was for years the warning “please make sure your schema is valid” went ignored, and we’d get too many bug reports of problems with invalid schemas. So I decided to just reach for what is the best schema validator I’ve personally used, and just tell people when their schemas are invalid.

Also as a bonus, we got to shed a lot of internal weight and maintenance and fix bugs by leveraging Redocly’s more mature schema resolver & bundler. But as a trade-off, Redocly makes the requests which means the previous methods of passing headers into the CLI didn’t work well (which we also had bug reports because it wasn’t very robust, and didn’t support all the different ways people protect their schemas—after all this isn’t a fetching client it’s a type generator).

For advanced cases, people would just have to download the file to a temp directory and then run the CLI on that anyway, so that’s also a common pattern for folks with very specific setups.

But all that said, if there are some easy CLI flags that would work with the new way in which schemas are fetched, we’d always welcome improvement PRs 🙂

**Reply by drwpow** [reply]
Oh and I should add—we don’t load the full Redocly CLI; just some of the lightweight core tools that are about as heavy as the original code was. So you don’t have to completely opt-in to Redocly when using this library (and there’s no telemetry sent, etc). You just get to use features of a more powerful, more mature ecosystem in this library if you need it.

**Reply by florianmartens** [reply]
Ah ok, awesome. Thanks for talking the time to explain this to me! Maybe, I'll find some time to look at the code as well. Would be exiting!
